src/control/server/mgmt_pool_test.go

```diff
@@ -1129,23 +1129,136 @@ func TestServer_MgmtSvc_PoolQuery(t *testing.T) {
+				HumanID: "nope-bad-not-gonna-work",
+			},
+			expErr: errors.New("unable to find pool service"),
+		},
+		"valid label lookup": {
+			req: &mgmtpb.PoolResolveIDReq{
+				HumanID: defaultLabel,
+			},
+			expResp: &mgmtpb.PoolResolveIDResp{
+				Uuid: mockUUID,
+			},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			log, buf := logging.NewTestLogger(t.Name())
+			defer common.ShowBufferOnFailure(t, buf)
+
+			ms := newTestMgmtSvc(t, log)
+			addTestPools(t, ms.sysdb, mockUUID)
+			ps, err := ms.sysdb.FindPoolServiceByUUID(uuid.MustParse(mockUUID))
+			if err != nil {
+				t.Fatal(err)
+			}
+			ps.PoolLabel = defaultLabel
+			if err := ms.sysdb.UpdatePoolService(ps); err != nil {
+				t.Fatal(err)
+			}
+
+			gotResp, gotErr := ms.PoolResolveID(context.TODO(), tc.req)
+			common.CmpErr(t, tc.expErr, gotErr)
+			if tc.expErr != nil {
+				return
+			}
+
+			if diff := cmp.Diff(tc.expResp, gotResp, common.DefaultCmpOpts()...); diff != "" {
+				t.Fatalf("unexpected response (-want, +got)\n%s\n", diff)
+			}
+		})
 	}
-	withNumVal := func(r *mgmtpb.PoolSetPropReq, v uint64) *mgmtpb.PoolSetPropReq {
-		r.SetValueNumber(v)
-		return r
+}
+
+func propWithName(r *mgmtpb.PoolSetPropReq, n string) *mgmtpb.PoolSetPropReq {
+	r.SetPropertyName(n)
+	return r
+}
+func propWithNumber(r *mgmtpb.PoolSetPropReq, n uint32) *mgmtpb.PoolSetPropReq {
+	r.SetPropertyNumber(n)
+	return r
+}
+
+func propWithStrVal(r *mgmtpb.PoolSetPropReq, v string) *mgmtpb.PoolSetPropReq {
+	r.SetValueString(v)
+	return r
+}
+func propWithNumVal(r *mgmtpb.PoolSetPropReq, v uint64) *mgmtpb.PoolSetPropReq {
+	r.SetValueNumber(v)
+	return r
+}
+
+func TestServer_MgmtSvc_PoolSetProp_Label(t *testing.T) {
+	defaultLabel := "test-label"
+
+	for name, tc := range map[string]struct {
+		label  string
+		expErr error
+	}{
+		"labels must be unique": {
+			label:  defaultLabel,
+			expErr: FaultPoolDuplicateLabel(defaultLabel),
+		},
+		"success": {
+			label: "unique-label",
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			log, buf := logging.NewTestLogger(t.Name())
+			defer common.ShowBufferOnFailure(t, buf)
+
+			ms := newTestMgmtSvc(t, log)
+			setupMockDrpcClient(ms, &mgmtpb.PoolSetPropResp{
+				Property: &mgmtpb.PoolSetPropResp_Number{
+					Number: drpc.PoolPropertyLabel,
+				},
+				Value: &mgmtpb.PoolSetPropResp_Strval{
+					Strval: tc.label,
+				}}, nil)
+			addTestPools(t, ms.sysdb, mockUUID)
+			ps, err := ms.sysdb.FindPoolServiceByUUID(uuid.MustParse(mockUUID))
+			if err != nil {
+				t.Fatal(err)
+			}
+			ps.PoolLabel = defaultLabel
+			if err := ms.sysdb.UpdatePoolService(ps); err != nil {
+				t.Fatal(err)
+			}
+
+			req := propWithStrVal(propWithName(new(mgmtpb.PoolSetPropReq), "label"), tc.label)
+			req.Uuid = mockUUID
+			_, gotErr := ms.PoolSetProp(context.TODO(), req)
+			common.CmpErr(t, tc.expErr, gotErr)
+			if tc.expErr != nil {
+				return
+			}
+
+			_, err = ms.sysdb.FindPoolServiceByLabel(tc.label)
+			if err != nil {
+				t.Fatal(err)
+			}
```

Comment:
Seems we should probably verify it's the correct pool as well.

Reply:
Done.